src/sidebar/components/test/thread-list-omega-test.js

```diff
@@ -0,0 +1,280 @@
+    fakeScrollContainer.className = 'js-thread-list-scroll-root';
+    fakeScrollContainer.style.height = '2000px';
+    document.body.appendChild(fakeScrollContainer);
+
+    fakeStore = {
+      clearSelection: sinon.stub(),
+    };
+
+    fakeTopThread = {
+      id: 't0',
+      annotation: { $tag: 'myTag0' },
+      children: [
+        { id: 't1', children: [], annotation: { $tag: 't1' } },
+        { id: 't2', children: [], annotation: { $tag: 't2' } },
+        { id: 't3', children: [], annotation: { $tag: 't3' } },
+        { id: 't4', children: [], annotation: { $tag: 't4' } },
+      ],
+    };
+
+    fakeVisibleThreadsUtil = {
+      calculateVisibleThreads: sinon.stub().returns({
+        visibleThreads: fakeTopThread.children,
+        offscreenUpperHeight: 400,
+        offscreenLowerHeight: 600,
+      }),
+      THREAD_DIMENSION_DEFAULTS: {
+        defaultHeight: 200,
+      },
+    };
+
+    $imports.$mock(mockImportedComponents());
+    $imports.$mock({
+      'lodash.debounce': fakeDebounce,
+      '../store/use-store': callback => callback(fakeStore),
+      '../util/annotation-metadata': fakeMetadata,
+      '../util/dom': fakeDomUtil,
+      '../util/visible-threads': fakeVisibleThreadsUtil,
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  it('works', () => {
+    const wrapper = createComponent();
+
+    assert.isTrue(wrapper.find('section').exists());
+  });
+
+  it('calculates visible threads', () => {
+    createComponent();
+
+    assert.calledWith(
+      fakeVisibleThreadsUtil.calculateVisibleThreads,
+      fakeTopThread.children,
+      sinon.match({}),
+      0,
+      sinon.match.number
+    );
+  });
+
+  context('new annotation created in application', () => {
+    it('attaches a listener for the BEFORE_ANNOTATION_CREATED event', () => {
+      fakeRootScope.$on = sinon.stub();
+
+      createComponent();
+
+      assert.calledWith(
+        fakeRootScope.$on,
+        events.BEFORE_ANNOTATION_CREATED,
+        sinon.match.func
+      );
+    });
+
+    it('clears the current selection in the store', () => {
+      createComponent();
+
+      fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED]({}, {});
+
+      assert.calledOnce(fakeStore.clearSelection);
+    });
+
+    it('does not clear the selection in the store if new annotation is a highlight', () => {
+      fakeMetadata.isHighlight.returns(true);
+      createComponent();
+
+      fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED]({}, {});
+
+      assert.notCalled(fakeStore.clearSelection);
+    });
+
+    it('does not clear the selection in the store if new annotation is a reply', () => {
+      fakeMetadata.isReply.returns(true);
+      createComponent();
+
+      fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED]({}, {});
+
+      assert.notCalled(fakeStore.clearSelection);
+    });
+  });
+
+  context('active scroll to an annotation thread', () => {
```

Comment:
After a bunch of hemming and hawing...as far as I can tell the only observable thing that happens in this second `useEffect` hook https://github.com/hypothesis/client/pull/2044/files#diff-0aa90e2743ab4cd906abb9caa7f3364bR78  is that the `scrollTop` for the scroll container is set.  Thus these tests culminate in checking what happens to that `scrollTop` property.

Reply:
Yep, that's the only externally visible behaviour that we're interested in.